src/numerics/petsc_vector.C

```diff
@@ -353,6 +375,9 @@ void PetscVector<T>::add (const T a_in, const NumericVector<T> & v_in)
 
   PetscErrorCode ierr = VecAXPY(_vec, a, v->vec());
   LIBMESH_CHKERR(ierr);
+
+  libmesh_assert(this->comm().verify(this->type()));
```

Comment:
nice

Reply:
Oh, but hell, I meant to put that in `add_vector()` too.  After looking at the MOOSE patch again I don't think this is the problem, but if it is let's find it at its root.